rust-driver-makefile.toml

```diff
@@ -1,66 +1,424 @@
+//! [dependencies]
+//! wdk-build = { path = "./crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "{}.sys",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
 '''
 
-[tasks.copypdb]
+[tasks.copy-pdb-to-package]
+private = true
 dependencies = ["build"]
+script_runner = "@rust"
+script_runner_args = [
+  "--base-path",
+  "${CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY}",
+]
 script = '''
-cd "%OUTPUT_DIR%"
-copy %CARGO_MAKE_CRATE_FS_NAME%.pdb package\%CARGO_MAKE_CRATE_FS_NAME%.pdb
+//! ```cargo
+//! [dependencies]
+//! wdk-build = { path = "./crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "{}.pdb",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
 '''
 
-[tasks.inf2cat]
+[tasks.copy-inf-to-package]
+private = true
 dependencies = ["stampinf"]
+script_runner = "@rust"
+script_runner_args = [
+  "--base-path",
+  "${CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY}",
+]
 script = '''
-inf2cat.exe /driver:"%OUTPUT_DIR%\package" /os:10_NI_X64,10_VB_X64 /uselocaltime /verbose
+//! ```cargo
+//! [dependencies]
+//! wdk-build = { path = "./crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "{}.inf",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
 '''
 
-[tasks.infverif]
-dependencies = ["stampinf"]
+[tasks.copy-map-to-package]
+private = true
+dependencies = ["build"]
+script_runner = "@rust"
+script_runner_args = [
+  "--base-path",
+  "${CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY}",
+]
 script = '''
-"%WDKToolRoot%\%Platform%\infverif.exe" /v /w "%OUTPUT_DIR%\package\%CARGO_MAKE_CRATE_FS_NAME%.inf" /msft
+//! ```cargo
+//! [dependencies]
+//! wdk-build = { path = "./crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "deps/{}.map",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
 '''
 
-[tasks.sign]
-dependencies = ["rename-dll-to-sys", "inf2cat", "infverif"]
+[tasks.inf2cat]
+private = true
+dependencies = ["copy-sys-to-package", "copy-inf-to-package"]
+command = "inf2cat"
+args = [
+  "/driver:${WDK_BUILD_OUTPUT_DIRECTORY}/${CARGO_MAKE_CRATE_FS_NAME}_package",
+  "/os:10_NI_X64,10_VB_X64",                                                   # TODO: this should be a parameter
+  "/uselocaltime",
+]
+
+[tasks.generate-certificate]
+private = true
+condition_script = [
```

Comment:
With https://github.com/sagiegurari/cargo-make/issues/1025 being resolved in cargo make v0.37.8, these condition scripts could be refactored to be rust-based for better maintainability, but since its fully functional right now, I'll leave that as a future improvement.